src/uagents/context.py

```diff
@@ -112,6 +130,29 @@ async def send(
             timeout=timeout,
         )
 
+    async def broadcast(
+        self,
+        destination: str,
+        message: Model,
+        timeout: Optional[int] = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
+    ):
+        agents = self.get_agents_by_protocol(destination)
+        if not agents:
+            self.logger.error(f"No active agents found for: {destination}")
+            return
+        schema_digest = Model.build_schema_digest(message)
+        for address in agents:
+            try:
+                await self.send_raw(
+                    address,
+                    message.json(),
+                    schema_digest,
+                    message_type=type(message),
+                    timeout=timeout,
+                )
+            except asyncio.exceptions.CancelledError:
+                self.logger.error(f"Error sending message to {address}")
```

Comment:
Potentially consider gathering all these awaitables together:

https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

Might make the error handling a little more complicated. Let me know what you think.

Reply:
Gather is a good idea I think as it should reduce complexity in the task queue. As for the error handling, this is a topic that should be tackled on a broader level first I think. James confirmed that this is something that was planned for the current/upcoming sprint, after that we can refine exception handling a layer above, e.g. in this broadcast method.